store/store.go

```diff
@@ -1587,7 +1601,30 @@ func (s *Store) snapConnCheck() ([]string, error) {
 	return hosts, nil
 }
 
+var ErrStoreOffline = errors.New("store is offline, use 'snap set system store.access=online' to go online")
+
+func (s *Store) isStoreOnline() error {
+	if s.dauthCtx == nil {
+		return nil
+	}
+
+	access, err := s.dauthCtx.StoreAccess()
+	if err != nil { // TODO: how to handle this error? some places log, some places bubble it up
+		return fmt.Errorf("cannot get store access from state: %w", err)
+	}
+
+	if access == "offline" {
+		return ErrStoreOffline
+	}
+
+	return nil
+}
+
 func (s *Store) ConnectivityCheck() (status map[string]bool, err error) {
```

Comment:
I think that's a good idea. Otherwise, it'll be hard to make sure we don't miss any now or in the future

Reply:
as we discussed separately one issue with that is that it might acess StoreOffline quite a bit which requires locking/unlocking the state

Reply:
Yes, but I thought it would be fine since the locking, parsing the JSON and unlocking would still be orders of magnitude faster than the actual HTTP request so, in the end, it won't have a significant effect in latency. If it does makes a significant difference, could we cache this in the store configuration? By having the store register a callback that the handler in configcore would call to update the configuration's value or to invalidate the cached value so `checkOnline` has to re-read it.

Reply:
@pedronis do you know of a good way to test out the performance impact of locking/unlocking the state for each HTTP request that comes from the store?

Also, as written right now, all public methods on the store that do anything over the network will have to acquire the lock at some point. If that happen in the HTTP client or not, does that make a difference? Unless, do some store operations result in multiple HTTP requests?

Reply:
I wouldn't agonize over this too much, most new methods that we would add medium term to the store we'll go through endpointURL, you might leave a TODO somewhere in the code about this issue

Reply:
Sound good, will do.